lib/ansible/vars/manager.py

```diff
@@ -344,6 +344,18 @@ def plugins_by_groups():
                 if not isinstance(vars_file_list, list):
                     vars_file_list = [vars_file_list]
 
+                # if 'vars_files' is directory, all files in dir are appended to vars_file_list.
```

Comment:
the code does not filter out 'forgein' files and assumes all files in the directory are 'yaml', i would instead refactor the host_group_vars plugin and/or include_vars code to a common source and use that.